base/acme/conf/engine.conf

```diff
@@ -3,4 +3,20 @@
 # above.
 #
 # Whether to enable the ACME service:
-enabled=1
+enabled=true
+
+# Whether to accept wildcard DNS identifiers:
+wildcard=true
+
+# Validity policies
+validities.nonce.length=30
+validities.nonce.unit=MINUTES
+
+validities.validAuthorization.length=30
+validities.validAuthorization.unit=MINUTES
```

Comment:
This PR presents an opportunity to review these defaults.  What do other ACME servers do?

Reply:
According to @cipherboy they possibly might not remove the records due to CT. Would it be a problem for IPA if the ACME database keeps growing?

Reply:
We should expire them.  We don't need to keep ACME objects around for CT - the CT log does that.  Considering high volume environments, I think we should default to pruning.

Reply:
RFC 8555 says in section 7.5.1:
```
The server SHOULD NOT remove challenges with status "invalid".
```
Currently we do not remove invalid challenges/authorizations/orders. Is that OK? Or should we do something different?

Reply:
Consider the whole paragraph:

```
   The server is said to "finalize" the authorization when it has
   completed one of the validations.  This is done by assigning the
   authorization a status of "valid" or "invalid", corresponding to
   whether it considers the account authorized for the identifier.  If
   the final state is "valid", then the server MUST include an "expires"
   field.  When finalizing an authorization, the server MAY remove
   challenges other than the one that was completed, and it may modify
   the "expires" field.  The server SHOULD NOT remove challenges with
   status "invalid".
```

My interpretation is that this requirement pertains to authz completion.  i.e. it is saying "challenges that were not attempted MAY be removed, but challenges that were attempted and happened to fail SHOULD kept, as SHOULD the challenge that succeeded".

In any case it's a SHOULD, and I am hesitant to keep objects forever, esp. invalid authorizations as it is a potential attack vector (flood the server with spurious orders and fail validations over and over).  Which reminds me, we should consider some rate limiting (is there a ticket for that?)

Reply:
I've created PR #495 to expire all records except accounts. Please take a look. If someone wants to retain the records they can disable the record validity policy.

I don't think we have a ticket yet for the rate limit. That would have to be done in the next release at the soonest.

Reply:
Sorry, not CT, but CC. CC requires an audit log with all issuances. If the order is in the terminal state, we should keep it so we know it was issued via ACME. I don't think I said authorizations necessarily need to be kept, but instead orders should be.

Reply:
In the future we might want to support event-driven data model, so all activities are recorded as events (including deletions) and we will never remove any records. This data model will work better in cloud. This is probably more appropriate for PostgreSQL, Cassandra, or Kafka, but LDAP database should continue to be supported for IPA.